runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/content/BigDecimal.kt

```diff
@@ -0,0 +1,9 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.smithy.kotlin.runtime.content
+
+public expect class BigDecimal(value: String) : Number {
+    public fun toPlainString(): String
```

Comment:
**Question**: Why?

Reply:
Ugh, because Java's `BigDecimal.toString()` returns a string in exponential notation (e.g., `1.23e20`) after the magnitude gets large enough. That's unsuitable for serialization because it would result in a loss of precision. So we need a way to get the full, unabridged decimal representation and JVM's version of that is [`toPlainString()`](https://docs.oracle.com/javase/8/docs/api/java/math/BigDecimal.html#toPlainString--). I don't love this, by the way—it feels conspicuous and strongly tied to a quirk of the JVM implementation.

Any suggestions for a more graceful way to handle it?

Reply:
I think thats fine if it's the accepted norm on JVM, its what most users will be using so makes sense to be consistent there than introduce a new concept.